subworkflows/local/prepare_input.nf

```diff
@@ -0,0 +1,64 @@
+nextflow.enable.dsl = 2
+
+import org.yaml.snakeyaml.Yaml
+
+workflow PREPARE_INPUT {
+
+    take:
+    ch_input
+
+    main:
+
+    ch_input.map { file -> readYAML( file ) }
+        .multiMap { file ->
+            samples_ch: file.samples
+        }
```

Comment:
```suggestion
        .map { file -> file.samples }
```
Alternatively you could have the `readYAML` function just return samples so there's no need for the extra `map` operation.